core/src/main/java/org/apache/brooklyn/core/config/BasicConfigInheritance.java

```diff
@@ -82,9 +223,17 @@ public InheritanceMode isInherited(ConfigKey<?> key, Object from, Object to) {
     
     protected <TContainer, TValue> void checkInheritanceContext(ConfigValueAtContainer<TContainer, TValue> local, ConfigInheritanceContext context) {
         ConfigInheritance rightInheritance = ConfigInheritances.findInheritance(local, context, this);
-        if (!equals(rightInheritance)) 
+        if (!isSameRootInstanceAs(rightInheritance)) {
             throw new IllegalStateException("Low level inheritance computation error: caller should invoke on "+rightInheritance+" "
                 + "(the inheritance at "+local+"), not "+this);
+        }
+    }
+
+    private boolean isSameRootInstanceAs(ConfigInheritance other) {
+        if (other==null) return false;
+        if (this==other) return true;
+        if (other instanceof DelegatingConfigInheritance) return isSameRootInstanceAs( ((DelegatingConfigInheritance)other).getDelegate() );
```

Comment:
`equals` would work but it wouldn't catch quite as many errors; it should be invoked on the same _instance_; have added comment to explain.  you're right dropping the delegate pattern would allow this line to be removed but i think losing the internal fields in persisted state is worth it.